syntheseus/reaction_prediction/inference/root_aligned.py

```diff
@@ -121,14 +121,15 @@ def _build_kwargs_from_scores(self, scores: List[float]) -> List[Dict[str, Any]]
 
             new_score = total_rr - (best_pos + 1) * max_possible_total_rr
             assert new_score <= 0.0
-            metadata = {
-                "original_score": score,
-                "best_pos": best_pos,
-                "total_rr": total_rr,
-                "score": new_score,
-            }
-
-            kwargs_list.append(metadata)
+
+            kwargs_list.append(
+                {
+                    "original_score": score,  # type: ignore[typeddict-unknown-key]
+                    "best_pos": best_pos,  # type: ignore[typeddict-unknown-key]
+                    "total_rr": total_rr,  # type: ignore[typeddict-unknown-key]
+                    "score": new_score,
+                }
+            )
```

Comment:
As expected, after improving the type hint `mypy` complains about the keys that `ReactionMetaData` does not know about. For now I just added appropriate `type: ignore` statements as it didn't seem useful to add RootAligned-specific keys to the underlying class, but I'm open to trying alternative approaches.

Reply:
Maybe make a plain dict and use `typing.cast` to force mypy to treat it as a `ReactionMetaData` object? That should hopefully work around whatever is happening with our python 3.7 tests.

Reply:
Actually just moving `# type: ignore[typeddict-unknown-key]` to apply to the entire dictionary (and not a single entry) seems to work 🙂